lib/roo_on_rails/logfmt.rb

```diff
@@ -0,0 +1,41 @@
+module RooOnRails
+  # A generator for the logfmt log format.
+  #
+  # @see https://brandur.org/logfmt The original description of logfmt
+  # @see https://godoc.org/github.com/kr/logfmt The 'reference' parser
+  module Logfmt
+    class << self
+      SPACE = 0x20
+      QUOTE = 0x22
+      EQUALS = 0x3d
+
+      def dump(hash)
+        return nil if hash.nil? || hash.empty?
+
+        hash.map { |k, v| "#{k}=#{dump_value(v)}" }.join(' ')
+      end
+
+      private
+
+      def dump_value(v)
+        str = case v
+              when String then v
+              when Symbol then v.to_s
+              else v.respond_to?(:to_json) ? v.to_json : v.inspect
+              end
+        escape(str)
```

Comment:
It's _really_ hard to write specs for this stuff and be sure you've got it right because you've got the conflict of Ruby's embedded escaping with the fact you actually want literally escaped characters in the output. I've been doing a whole bunch of testing by using it in an app and making sure the output is correct in the real log output too ;-)